scipy/linalg/tests/test_decomp_polar.py

```diff
@@ -91,3 +92,29 @@ def test_verify_cases():
     for a in verify_cases:
         verify_polar(a)
 
+@pytest.mark.parametrize('dt', [int, float, np.float32, complex, np.complex64])
+def test_empty(dt):
+    empty_cases = [
+    np.empty((0, 0), dt),
+    np.empty((0, 2), dt),
+    np.empty((2, 0), dt),
+    ]
+
+    for a in empty_cases:
+        # a = up 
+        u, p = polar(a, side='right')
+        u_n, p_n = polar(np.eye(2, dtype=dt))
+
+        assert_equal(u.dtype, u_n.dtype)
+        assert_equal(p.dtype, p_n.dtype)
+        assert_(u.size == 0)
+        assert_((p == 0).all())
+
+        # a = pu
+        u, p = polar(a, side='left')
+        u_n, p_n = polar(np.eye(2, dtype=dt))
+
+        assert_equal(u.dtype, u_n.dtype)
+        assert_equal(p.dtype, p_n.dtype)
+        assert_(u.size == 0)
+        assert_((p == 0).all())
```

Comment:
I can't easily assess whether this is the correct numerical value. I will either investigate or ask for another review before merging.

Reply:
> p is Hermitian positive semidefinite. If a is nonsingular, p is positive definite.

A square array of zeros is Hermitian positive semidefinite. I don't think `a` can be "nonsingular" if it is not square, e.g. `(0, 5)` or `(5, 0)`, so in these cases, it is OK that `p` is not strictly positive definite.  When `a` is square of shape `(0, 0)`, `p` will also be of shape `(0, 0)`, so I think we're good here.